src/api-binder/report.ts

```diff
@@ -70,30 +76,44 @@ async function report({ body, opts }: StateReport) {
 	}
 }
 
+/**
+ * Collects current state and reports with backoff. Diffs report content with
+ * previous report and does not send an empty report.
+ *
+ * Does *not* validate time elapsed since last report.
+ */
 async function reportCurrentState(opts: StateReportOpts) {
-	// Wrap the report with fetching of state so report always has the latest state diff
 	const getStateAndReport = async () => {
-		const now = performance.now();
-		// Only try to report if enough time has elapsed since last report
-		if (now - lastReportTime >= constants.maxReportFrequency) {
-			const currentState = await deviceState.getCurrentForReport(lastReport);
-			const stateDiff = prune(shallowDiff(lastReport, currentState, 2));
```

Comment:
Note that this trades off device resources for API resources, since `getCurrentForReport` incurrs a non-negligible cost. I think this is the right tradeoff to make as the API is much harder to scale, but I just wanted to point it out.

Reply:
Sorry, you are right, the snippet is weird. And actually I'm wrong, I was thinking `getCurrentForReport` now would be called more often on account of the removal of the `if` clause before, but I had forgot that you moved that comparison further up the stack. Disregard this then